### PR TITLE
BugFix Issue #262 Prevent cron from paying deposit invoice with credits

### PR DESCRIPTION
**Pull request by DataLores**
Deposit invoices types are generated during client add funds and should only be payable via payment gateways with valid transactions.

### CODE REVIEW

File: src/bb-modules/Invoice/ServiceInvoiceItem.php [hunk 1]
```diff
@@ -121,16 +121,8 @@ public function executeTask(\Model_InvoiceItem $item)
         }
 
         if (\Model_InvoiceItem::TYPE_DEPOSIT == $item->type) {
-            $clientService = $this->di['mod_service']('Client');
-
-            $invoice = $this->di['db']->getExistingModelById('Invoice', $item->invoice_id);
```
Comment (jaapmarcus) [inline src/bb-modules/Invoice/ServiceInvoiceItem.php:126]: Due to changes here the test seems to be failing

1) Box\Mod\Invoice\ServiceInvoiceItemTest::testexecuteTaskTypeDeposit
[44](https://github.com/FOSSBilling/FOSSBilling/actions/runs/3062942591/jobs/4949750766#step:10:45)
Expectation failed for method name is "getExistingModelById" when invoked at least once.
[45](https://github.com/FOSSBilling/FOSSBilling/actions/runs/3062942591/jobs/4949750766#step:10:46)
Expected invocation at least once but it never occurred.

It seems this function is used for automated testing to add "deposit" to the user

**Review by jaapmarcus — CHANGES_REQUESTED** [review]

File: src/bb-modules/Invoice/ServiceInvoiceItem.php (hunk 1 above)
Comment (jaapmarcus) [reply src/bb-modules/Invoice/ServiceInvoiceItem.php:126]: Or we need to update ...

**Review by evrifaessa — APPROVED** [review]
LGTM. Was able to reproduce the problem and confirm the fix.

**Review by jaapmarcus — APPROVED** [review]

**Comment by jaapmarcus** [comment]
👍🏼  Thanks

**Pull request merged**